### ISSUE TITLE
Enforce AHUB Policy

### ISSUE THREAD

**Issue by bennbatt8112** [Azure/azure-policy#21]
Hi,

FTE here, feel free to reach out. Wanted to provide some general feedback as well as some feedback/raise an issue specific to the Enforcing AHUB policy.

I tested the Enforce AHUB policy in a subscription of mine and ran into an issue where Linux distros could no longer be deployed as they do not have the Microsoft.Compute/LicenseType attribute. My understanding is that Microsoft.Compute/LicenseType will return $null, and $null _IS_ in fact not equal to "Windows_Server" which will register as $true and deny the build. I see a few way to get around this, 1) adding a condition that Microsoft.Compute/LicenseType notequals $null, or 2) adding some logic to ensure the ImageOffer is also "WindowsServer" or "WindowsServerSemiAnnual. These aren't fully flushed out, as certain offer types, like Nano server are contained, but don't offer AHUB. I've attached a work around that I use, however I manually checked which Microsoft Published Offers/Skus had AHUB available. Hoping I can get help vetting and testing a more flushed out solution that doesn't block Linux VMs.

As for general feedback, Azure Policy is a super powerful tool, but more documentation and PowerShell options would be really helpful here. It seems like there is a bit of overlap in Aliases within policy, e.g. Microsoft.Compute/imageSku & Microsoft.Compute/virtualMachines/imageSku. Mapping the aliases to REST API attributes might be helpful. This led to some of my confusion with modifying the sample AHUB template. Also would be useful to provide some cmdlets regarding policy initiatives, as I only see the option to create these in the Portal. Would be happy to help in this effort in any way possible or answer/ask additional questions.

[sampleAHUBDeny.txt](https://github.com/Azure/azure-policy/files/1722246/sampleAHUBDeny.txt)

**Comment by cyl33922072** [comment]
I fixed the issue by adding the image condition. Can you verify it?

**Comment by bennbatt8112** [comment]
Hey,

Thanks for the response. So I ran just a couple tests and I think your policy solves the issue of denying Linux VMs.

The only thing I would add is that the updated policy as it stands does not enforce the WindowsServerSemiAnnual Images which are also eligible for AHUB. Right now the Windows Server Version 1709 and I believe one other are eligible for AHUB but not denied here.

Wish there were a list that could be referenced so the policy was updated as AHUB eligibility is updated.
